src/benchmarks/micro/corefx/System.IO.Pipelines/PipeThroughputBenchmark.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public class PipeThroughputBenchmark
+    {
+        private const int InnerLoopCount = 512;
+
+        private PipeReader _reader;
+        private PipeWriter _writer;
+        private MemoryPool<byte> _memoryPool;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _memoryPool = KestrelMemoryPool.Create();
```

Comment:
I think we can use the default pool and drop dependency on kestrel

Reply:
@pakrym good point, I have changed it to use the Shared one instead.